include/rdma/fi_ext.h

```diff
@@ -123,6 +124,49 @@ struct fid_mem_monitor {
 	struct fi_ops_mem_notify *import_ops;
 };
 
+/*
+ * System logging import extension:
+ * To use, open logging fid and import.
+ */
+
+struct fi_ops_log {
+	size_t size;
+	int (*enabled)(int disable, enum fi_log_level level,
+		       enum fi_log_subsys subsys);
```

Comment:
`fi_provider` isn't defined in any header the application could include. Shall the application consider it as an opaque structure?
Also, I don't quite understand how to set the flags: could you please give me an example?

Reply:
Hmm... do we not install the files in rdma/providers? Doesn't look like it but maybe we should.  The APIs and structures in those files are stable, and it would be safe to read fi_provider::name, for example.

For the flags example:
```
/* in fi_ext.h */
#define FI_LOG_PROV_FILTERED (1ULL << 0)
```
log_enabled(prov, ..., flags)
flags - if FI_LOG_PROV_FILTERED flag has been set, the user has requested that the specified provider (prov parameter) be filtered from logging output.

The app can use the flags to filter output or override the filtering mechanism with their own.  There could be additional flags for subsystem and level filtering.

If we don't want to install the rdma/providers files, we could pass the provider name in place of the fi_provider structure.